docs-client/src/components/GotoSelect/index.tsx

```diff
@@ -0,0 +1,328 @@
+    },
+    paper: {
+      position: 'absolute',
+      zIndex: 1,
+      marginTop: theme.spacing.unit,
+      left: 0,
+      right: 0,
+      // width: '125%',
+      flex: 1,
+    },
+    divider: {
+      height: theme.spacing.unit * 2,
+    },
+  });
+
+function NoOptionsMessage(props: NoticeProps<OptionType>) {
+  return (
+    <Typography
+      color="textSecondary"
+      className={props.selectProps.classes.noOptionsMessage}
+      {...props.innerProps}
+    >
+      No Results
+    </Typography>
+  );
+}
+
+type InputComponentProps = Pick<BaseTextFieldProps, 'inputRef'> &
+  HTMLAttributes<HTMLDivElement>;
+
+function inputComponent({ inputRef, ...props }: InputComponentProps) {
+  return <div ref={inputRef} {...props} />;
+}
+
+function Control(props: ControlProps<OptionType>) {
+  const {
+    children,
+    innerProps,
+    innerRef,
+    selectProps: { classes, TextFieldProps },
+  } = props;
+
+  return (
+    <TextField
+      fullWidth
+      InputProps={{
+        inputComponent,
+        inputProps: {
+          children,
+          className: classes.input,
+          ref: innerRef,
+          ...innerProps,
+        },
+        disableUnderline: true,
+      }}
+      {...TextFieldProps}
+    />
+  );
+}
+
+function Option(props: OptionProps<OptionType>) {
+  return (
+    <MenuItem
+      buttonRef={props.innerRef}
+      selected={props.isFocused}
+      component="div"
+      style={{
+        fontWeight: props.isSelected ? 500 : 400,
+        width: 800,
+      }}
+      {...props.innerProps}
+    >
+      {props.children}
+    </MenuItem>
+  );
+}
+
+function Placeholder(props: PlaceholderProps<OptionType>) {
+  return (
+    <Typography
+      color="textSecondary"
+      className={props.selectProps.classes.placeholder}
+      {...props.innerProps}
+    >
+      Go to ...
+    </Typography>
+  );
+}
+
+function SingleValue(props: SingleValueProps<OptionType>) {
+  return (
+    <Typography
+      className={props.selectProps.classes.singleValue}
+      noWrap={true}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Typography>
+  );
+}
+
+function ValueContainer(props: ValueContainerProps<OptionType>) {
+  return (
+    <div className={props.selectProps.classes.valueContainer}>
+      {props.children}
+    </div>
+  );
+}
+
+function Menu(props: MenuProps<OptionType>) {
+  return (
+    <Paper
+      square
+      className={props.selectProps.classes.paper}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Paper>
+  );
+}
+
+const components = {
+  Control,
+  Menu,
+  NoOptionsMessage,
+  Option,
+  Placeholder,
+  SingleValue,
+  ValueContainer,
+};
+
+function makeSuggestions(specification: Specification): GroupType[] {
+  const suggestions: GroupType[] = [];
+
+  if (specification.getServices().length > 0) {
+    suggestions.push({
+      label: 'Services',
+      options: specification.getServices().flatMap((service) =>
+        service.methods.map((method) => {
```

Comment:
How about limiting the number of items shown in the suggestions, just in case we have too many of them?

Reply:
It seems good to limit the numbers of showing items. I didn't catch it. :-)
But I think users got confused when the items which exist are not shown just after typing.

I guess the memory footprint of suggestions is almost similar to specifications. But the items which are shown on screen have an impact on memory usage and rendering time.

What about the logic below?
* To limit the initial number of items around 100 for up|down key navigation.
* To reload items until 20 when filtering the items(typing in the control).

Reply:
Sounds good to me.